src/instrumentation/express/error_handler.ts

```diff
@@ -0,0 +1,31 @@
+import { Request, Response, NextFunction, ErrorRequestHandler } from "express"
+import * as opentelemetry from "@opentelemetry/api"
+
+export function expressErrorHandler(): ErrorRequestHandler {
+  return function (
+    err: Error & { status?: number },
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ) {
+    const activeSpan = opentelemetry.trace.getSpan(
+      opentelemetry.context.active()
+    )
+    // if there's no `status` property, forward the error
+    // we also ignore client errors here
+    if (
+      activeSpan &&
+      err &&
+      (!err.status || (err.status && err.status >= 500))
```

Comment:
Curious: who sets this `status` property on the Error object? Is this an Express thing? Can't find it on their docs.

Reply:
Yes! It's "mentioned" here: https://expressjs.com/en/guide/error-handling.html#the-default-error-handler

_The res.statusCode is set from err.status (or err.statusCode). If this value is outside the 4xx or 5xx range, it will be set to 500._

`status` is a Express alias for `statusCode` that is added by node to HTTP errors.

That's why we "enrich" the type with: `err: Error & { status?: number }`

Reply:
Huh. Thanks! Should we also be looking at `statusCode` then?

Reply:
I don't think so, we should be safe using `status` as Express docs recommend. If we were outside of Express, yes, we should be looking at `statusCode`